merge upstream/stable & bump v0.9.7 centos/8

necro-nemesis opened the pull request:
minor formatting for this purpose.

majestrate committed:
refactor logging to use std::source_location

* use std::source_location instead of godawful macros in logging
* remove unused/absolutely haram af json logstream
* fix bug in android logger where it doesn't respect eLogNone

Commit:
make source location happy on macos
* because of course apple doesn't provide any implementation (lmao) we provide one ourself

majestrate committed:
use inline source_location implementation for android too

majestrate committed:
for some reason [[maybe_unused]] is hated by buster 32 bit x86 so we do something else
to appease that compiler.

majestrate committed:
bump ngtcp2 submodule

majestrate committed:
make it compile again (squashme)

majestrate committed:
make lokinet-bootstrap target toggleable

add cmake option to disable building lokinet-bootstrap binary

jagerman committed:
Redo how we build ngtcp2 lib

ngtcp2's top-level CMakeLists.txt is not friendly to being used as a
subdirectory (for instance, it always adds a `check` target when cunit
is installed), so stop using it in favour of skipping directly into the
ngtcp2/lib subdir.

This requires some hackery to set up a couple of the things the
top-level cmake does, but it isn't *too* painful.

majestrate committed:
Merge pull request #1736 from jagerman/ngtcp2-only-lib

Redo how we build ngtcp2 lib

jagerman committed:
Apply suggestions from code review

jagerman committed:
Add mainnet/testnet bootstrap files into repo

We use them everywhere now (include the debs) so it makes sense to have
them bundled here rather than needing a download during package build
(in particular for debs that's considered a bad thing).

majestrate committed:
Merge pull request #1735 from majestrate/cmake-lokinet-bootstrap-tweaks-2021-09-10

make lokinet-bootstrap target toggleable

majestrate committed:
Update exit-setup.md

`hops=1` is not supported anymore.

jagerman committed:
Remove duplicate handling of jemalloc

We were linking/loading it in different ways, one with cmake option
`USE_JELLOC` and the other, older version `WITH_JEMALLOC`.  This removes
the latter (which was default OFF) and keeps the former (which was added
and has been default ON since 0.9.4 or so).

Also removes the `ifdef`ed JEMALLOC code in lokinet.cpp because we don't
need it; just linking to jemalloc is enough to get the malloc/free
replacements.

majestrate committed:
Merge pull request #1738 from jagerman/fix-jemalloc-option-dupe

Remove duplicate handling of jemalloc

majestrate committed:
Update readme.md

our readme sucks, make it suck a bit less

majestrate committed:
Update readme.md

more updates to readme, remove unmaintained parts, add and remove chunks of information.

majestrate committed:
Update readme.md

majestrate committed:
Update readme.md

majestrate committed:
Update readme.md

majestrate committed:
SystemD DNS fix:

when dns is port 53 call SetLinKDNS otherwise call SetLinkDNSEx as on older versions of systemd-resolved SetLinkDNSEx is not available.

majestrate committed:
Merge pull request #1741 from majestrate/systemd-resolved-dns-fix-2021-09-16

SystemD DNS fix

majestrate committed:
make lokinet-util library static

every other lokinet internal lib is static, but liblokinet-util was not, this fixes that.

majestrate committed:
fix up gripes in source_location

* get rid of columns we dont need those

majestrate committed:
refactor for apple

majestrate committed:
update log statement for win32

majestrate committed:
use llarp:: prefix for LogError

majestrate committed:
fix up logging, put _log into anonynous namespace, make only log trace nop in release builds

majestrate committed:
update submodule

majestrate committed:
make it compile again

majestrate committed:
update ci to use main lokinet mobile repo

jagerman committed:
Merge pull request #1750 from majestrate/ci-update-2021-09-24

update ci to use main lokinet mobile repo

jagerman committed:
Merge pull request #1751 from majestrate/update-ngtcp2-2021-08-24

Update ngtcp2

majestrate committed:
always build ngtcp2 as static

squashme

majestrate committed:
some compilers hated [[maybe_unused]] so use (void) when they are unused

majestrate committed:
Merge pull request #1756 from majestrate/ngtcp2-build-static-fix-2021-10-06

always build ngtcp2 as static

majestrate committed:
iterate through directory entries in lexigraphical order

majestrate committed:
Update fs.hpp

style nit

majestrate committed:
Merge pull request #1638 from majestrate/log-refactor-source-location-2021-05-11

refactor logging to use std::source_location

majestrate committed:
Merge pull request #1759 from majestrate/iterate-dir-in-order-2021-10-06

iterate through directory entries in lexigraphical order

majestrate committed:
handle exceptions on setup and config load without aborting

majestrate committed:
set base v6 address to nullopt when explicit empty string is provided

majestrate committed:
Merge pull request #1761 from majestrate/v6-base-address-configuration-fix-2021-10-06

resepct empty ipv6 range setting

majestrate committed:
catch ipv6 ioctl throwing when someone runs lokinet with no ipv6

fyrak1s committed:
Add missing dep 'cppzmq' (#1764)

* Add missing dep to cppzmq

majestrate committed:
prevent bizare half open state. (#1754)

* attempt path timeout bullshittery fix

* make sure ServiceInfo always has its address set up

* do not copy intros in constuctor, ammend logging and
add assert

jagerman committed:
Merge pull request #1762 from majestrate/v6-base-address-configuration-fix-2021-10-06

catch ipv6 ioctl throwing when someone runs lokinet with no ipv6

jagerman committed:
Merge pull request #1760 from majestrate/default-to-unix-socket-for-rpc-2021-10-06

catch exceptions on setup and dont abort

darcys22 committed:
RPC call for summary get_status (#1742)

* RPC call for summary get_status

* lint

* update with review notes

* further review points

* uint64_t

majestrate committed:
fix debian sid compiler error

jagerman committed:
Update clang build to 13; add full llvm-13 build

jagerman committed:
Add missing deprecated copy assignment operator

Clang-13 warns (and -Werror dies) without it.

jagerman committed:
Add jsonnet formatting (if jsonnet is installed)

jagerman committed:
make format

jagerman committed:
which => command -v; different format.sh exit codes

`which` is a debian tool that is being deprecated in favour of
posix-standard `command -v`, so which to that to avoid deprecation
warnings.

Change the exit codes of `contrib/format.sh verify` to be unique for
each formatting program.

jagerman committed:
Remove windows linefeeds

necro-nemesis committed:
formatting

necro-nemesis committed:
Update .drone.jsonnet

necro-nemesis committed:
formatting

necro-nemesis committed:
remove Patch2: default-upstream-dns.patch

necro-nemesis commented:
Patch if still necessary requires refreshing but it appears that /llarp/config/config.cpp now incorporates these changes to upstream dns. Patch2: /SOURCES file currently retained for reference.

jagerman committed:
Add libzmq3-dev; change deps into arrays

We're currently rebuilding libzmq3-dev all the time in most of the CI
jobs.

jagerman commented on SPECS/lokinet.spec:
```diff
@@ -24,7 +24,7 @@ BuildRequires:  libsqlite3x-devel
 Patch1: version-as-rpm-version.patch
 # Backport default upstream dns not working from PR 1715:
-Patch2: default-upstream-dns.patch
+# Patch2: default-upstream-dns.patch
```
You should just delete this line rather than commenting it out, and delete the patch file itself from the repository.

necro-nemesis replied:
Removed. ci failed and required manual restart to build successfully. Building on centos this seems to be a common occurrence.

necro-nemesis committed:
remove commented patch

Commit:
ngtcp2 submodule update

Commit:
Merge remote-tracking branch 'origin/dev' into centos/8

necro-nemesis committed:
.gitignore common

add !/SOURCES/bootstrap.signed and retain !/contrib/bootstrap/*.signed

jagerman committed:
Docker image updates

- split debian sid into base/sid/clang images
- similarly for debian stable
- add jsonnet to lint
- add `--pull` to docker build so that we always pull the latest images
  (otherwise we were building on whatever local cache we have for
  `debian:sid`, etc., which made the base image update layer much
  larger).
- don't install Recommends by default
- add libzmq3-dev
- split android into android (base) and flutter
- hard-code registry.oxen.rocks into the dockerfile stuff because that
  seems to be the only way to properly depend on other docker builds.
- update a few CI builds that should have been using our images but
  weren't.
- Update a few CI images to bullseye instead of buster

Add openssh-client (for sftp to upload builds)

jagerman committed:
Another gcc-11 fix

GCC is wrongly warning that `h` is uninitialized here, but it clearly
isn't.  Work around it.

jagerman committed:
Merge pull request #1769 from majestrate/fix-sid-2021-10-13

fix debian sid compiler error

jagerman committed:
Update deb repo dns name

jagerman committed:
CI docker multiarch

Adds multiarch docker images and starts using them.

jagerman committed:
Docker package updates:

- add `make`
- add `patch`

jagerman committed:
Reformat RUN commands to make editing easier

- split them on multiple lines
- sort the installed package deps
- don't use extra unnecessary `bash` invocation

jagerman committed:
Update Catch2 to latest to fix build on impish

jagerman committed:
Add automake, libtool, libjemalloc-dev

jagerman committed:
Add cached images with debhelper & related tools for deb pipelines

jagerman committed:
No recommends

jagerman committed:
Make sure we always pull the latest image

jagerman committed:
Fix arch-specific docker repo names

jagerman committed:
Disable ABI warnings on buster/armhf

jagerman committed:
Add missing xz-utils to buster

jagerman committed:
Reorder pipelines to put static builds together

majestrate committed:
Merge pull request #1774 from jagerman/clang-13

Docker + clang updates

jagerman committed:
Fetch CI submodules in parallel

jagerman committed:
Merge pull request #1775 from jagerman/parallel-submodules

Fetch CI submodules in parallel

majestrate committed:
version bump to 0.9.7

jagerman committed:
Merge pull request #1776 from oxen-io/dev

v0.9.7

necro-nemesis committed:
Merge remote-tracking branch 'upstream/stable' into centos/8

necro-nemesis committed:
bump version
